tests/extmod_hardware/machine_uart_irq_break.py

```diff
@@ -0,0 +1,67 @@
+# Test machine.UART.IRQ_BREAK firing after a break is received.
+#
+# IMPORTANT: This test requires hardware connections: the UART TX and RX
+# pins must be wired together.
+
+try:
+    from machine import UART
+
+    UART.IRQ_BREAK
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+import time, sys
+
+# Configure pins based on the target.
+if "esp32" in sys.platform and not (
+    "ESP32S2" in sys.implementation._machine or "ESP32C3" in sys.implementation._machine
```

Comment:
I suggest to SKIP s2/s3 like this:
```py
if "esp32" in sys.platform:
    if "ESP32S2" in sys.implementation._machine or "ESP32C3" in sys.implementation._machine:
        print("SKIP")
        raise SystemExit
    ... set up pins for esp32
elif "rp2" in sys.platform:
    ... set up pins for rp2
else:
    print("Please add support for this test on this platform.")
    raise SystemExit
```
That way any new ports in the future that support IRQ_BREAK have an appropriate warning about adding support to this test.